### ISSUE TITLE
[Feature Request] Specify hover point of an element

### ISSUE THREAD

**Issue by ris58h** [puppeteer/puppeteer#3146]
Currently `hover` methods place mouse over the center of an element. It could be useful to have an ability to specify a point of the hovered element. For example `page.hover('#id', { top: 0, left: 0})`.

**Comment by aslushnikov** [comment]
@ris58h what's the use case?

**Comment by ris58h** [comment]
When two elements intersect each other and center of underlying element is hovered by overlying one. Maybe, it's not that common, but I've had to this (I've had to write some code to get round this) and, in my opinion, having such parameter in `hover` method would look natural. Anyway it's just a proposal.

**Comment by realtaraharris** [comment]
If you're using this to hover on an SVG element as I am, you might want to be able to move the hover point a few pixels here or there so that it doesn't shoot right through the holes in the element. I'm not certain that's what's happening with the issue that brought me here, but it's a possibility I have not ruled out yet.

As an aside, I really wish puppeteer had a built-in way of showing you where the hell clicks are actually happening - something like a mode where it draws a persistent dot on each click. I ended up writing some code that does in my test harness, which was cool for me, but it's not gonna help anyone else...

EDIT: Aha! I found https://github.com/GoogleChrome/puppeteer/blob/master/test/assets/input/mouse-helper.js, which is nice, but why not make this an actual feature so people don't end up cloning that code into their own repos? This is something I've needed in every single puppeteer test I've ever written.
